### DISCUSSION TITLE
Calculate Health Factor Test

### DISCUSSION THREAD

**Discussion by mgroovyank** [Cyfrin/foundry-full-course-cu#2311] [Q&A] [answered]
Here is a test that we are performing:

```
    function testRevertHealthFactorBrokenDepositCollateralAndMintDSC() public {
        vm.startPrank(USER);
        ERC20Mock(weth).approve(address(dscEngine), AMOUNT_COLLATERAL);
        // 10 ether * 2000 USD/ETH = 20000 USD
        // 10000 USD - max dsc mint limit
        //  10000/200*100 = 10000/20000 = 0.5
        uint256 expectedHealthFactor =
            dscEngine.calculateHealthFactor(dscEngine.getUSDValue(weth, AMOUNT_COLLATERAL), 200 * DSC_MINT_AMOUNT);
        vm.expectRevert(abi.encodeWithSelector(DSCEngine.DSCEngine__BreaksHealthFactor.selector, expectedHealthFactor));
        dscEngine.depositCollateralAndMintDSC(weth, AMOUNT_COLLATERAL, 200 * DSC_MINT_AMOUNT);
        vm.stopPrank();
    }
```
In this test we use calculateHealthFactor function to get expected health factor. Is this the right approach for testing?
If we utilize the written smart contract code itself to match expected and actual values, we might miss some calculation mistake that we made in our contract. 

Wouldn't it be better if we perform calculation manually to get expected health factor? And then try to match it with the response from contract function?

**Comment by EngrPips** [comment, marked as answer]
Hello @mgroovyank, Yeah, you are kind of right, but it is not as if the approach of getting the values from the written contract in our test to match what we expected is totally wrong because most time, we test the behavior of our contract rather than values so intuitively when writing a contract you know what you want it to do and you can easily tell if it is not doing it but yeah the idea of doing calculations manually we surely be an extra layer of protection to ensure that truly we are doing what is correct and our contract is performing appropriately.
